daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
                   result = Success(Right(results.toSeq))
               }
           }
+        case SResultFinalValue(v) =>
+          // The final result should always be unit.
+          result = Failure(new RuntimeException(s"Unexpected non-unit final result: $v"))
         case SResultScenarioCommit(_, _, _, _) =>
           result = Failure(
             new RuntimeException("FATAL: Encountered scenario commit in DAML Script"))
         case SResultError(err) =>
           // Capture the error and exit.
           result = Failure(err)
-        case err =>
-          // TODO: Figure out when we hit this
-          // Capture the error (but not as SError) and exit.
-          result = Failure(new RuntimeException(s"FAILED: $err"))
+        case SResultNeedTime(callback @ _) =>
```

Comment:
What happens if you have a `getTime` call in a choice that you call via DAML Script? I would expect that you will hit this in this case. The one handled by the DAML Script runner is `getTime` outside of a transaction.

Reply:
Indeed, I've extended the time test-case and changed this bit to handle this case.